scripts/actions/spells/trust/prishe_ii.lua

```diff
@@ -12,7 +12,58 @@ spellObject.onSpellCast = function(caster, target, spell)
 end
 
 spellObject.onMobSpawn = function(mob)
-    xi.trust.message(mob, xi.trust.messageOffset.SPAWN)
+    local psychoAnima  = 3539
+
+    xi.trust.teamworkMessage(mob, {
+        -- Offsets for Lion II and Iroha II messages appear to be incorrect.
+        [xi.magic.spell.NASHMEIRA_II] = xi.trust.messageOffset.TEAMWORK_2,
+        [xi.magic.spell.LILISETTE_II] = xi.trust.messageOffset.TEAMWORK_3,
+        [xi.magic.spell.ARCIELA_II] = xi.trust.messageOffset.TEAMWORK_4,
+    })
+
+    local itemOneGambit = mob:addSimpleGambit(ai.t.SELF, ai.c.HPP_LT, 35, ai.r.MS, ai.s.SPECIFIC, psychoAnima)
+    -- TODO: Add additional logic for Hysteroanima
+    -- local itemTwoGambit = mob:addSimpleGambit(ai.t.TARGET, ai.c.CASTING_MA, 35, ai.r.MS, ai.s.SPECIFIC, 3540)
+    local healingGambit = mob:addSimpleGambit(ai.t.PARTY, ai.c.HPP_LT, 25, ai.r.MA, ai.s.HIGHEST, xi.magic.spellFamily.CURAGA)
+
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.STATUS, xi.effect.SLEEP_I, ai.r.MA, ai.s.SPECIFIC, xi.magic.spell.CURAGA)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.STATUS, xi.effect.SLEEP_II, ai.r.MA, ai.s.SPECIFIC, xi.magic.spell.CURAGA)
+
+    -- Trust Synergy: If Ulmia in the party, Prishe will cast Curaga when party members are in the yellow instead of red.
+    -- TODO: Prishe will use Cure spells exclusively on Ulmia (Only Cure I-IV)
+    -- https://ffxiclopedia.fandom.com/wiki/Trust:_Prishe_II
+    mob:addListener('COMBAT_TICK', 'PRISHE_CTICK', function(mobArg)
```

Comment:
Gambit system is designed specifically around the idea of not attaching any logic to a Trust's tick. I'd rather you leave a TODO here for this and I'll add a binding for "TrustAddedToParty" or something for you to use later

Reply:
You can approximate bindings for this with this diff:
```diff
diff --git a/scripts/globals/trust.lua b/scripts/globals/trust.lua
index 26746347e5..49296a8b68 100644
--- a/scripts/globals/trust.lua
+++ b/scripts/globals/trust.lua
@@ -267,6 +267,10 @@ xi.trust.onTradeCipher = function(player, trade, csid, rovCs, arkAngelCs)
     end
 end
 
+xi.trust.handleSynergyWithOtherTrusts = function(trust)
+
+end
+
 xi.trust.canCast = function(caster, spell, notAllowedTrustIds)
     -- Trusts must be enabled in settings
     if xi.settings.main.ENABLE_TRUST_CASTING == 0 then
@@ -380,13 +384,15 @@ xi.trust.canCast = function(caster, spell, notAllowedTrustIds)
 end

 xi.trust.spawn = function(caster, spell)
-    caster:spawnTrust(spell:getID())
+    local trust = caster:spawnTrust(spell:getID())

     -- Records of Eminence: Call Forth an Alter Ego
     if caster:getEminenceProgress(932) then
         xi.roe.onRecordTrigger(caster, 932)
     end

+    xi.trust.handleSynergyWithOtherTrusts(trust)
+
     return 0
 end
```

But I haven't thought of a _clean and readable_ way to lay out and track the synergies with other trusts yet, since some of them are singular, some of them are grouped, some add effects to some/all of the group, some add/remove gambits etc.

Open to suggestions

Reply:
Good examples of synergies:
https://ffxiclopedia.fandom.com/wiki/Trust:_Curilla and https://ffxiclopedia.fandom.com/wiki/Trust:_Rainemard
https://ffxiclopedia.fandom.com/wiki/Trust:_Trion and https://ffxiclopedia.fandom.com/wiki/Trust:_Pieuje_(UC)
and I forget where it mentions that some combination of Lion+Zeid+Tenzen each get different things if they're up at the same time

Reply:
Each synergy will need to be wrapped in something that can apply and remove it, since the synergy can/will be broken at any point if a trust is dismissed or dies. If each synergy is a table, then it's fine for the first entry to be a function that applies all the mods/actions/listeners whatever, and then the second entry removes them. In this layout, we'd make sure to remove all synergies every time the function is "tapped", to make sure we're starting on a clean slate. Then trying to re-apply.

Reply:
I'll try to make sense of this. I pulled this example from Mumor, here https://github.com/LandSandBoat/server/blob/17498511563b0da5175f368ae65ade0bfc703b4b/scripts/actions/spells/trust/mumor.lua#L30 and it
seems to work rather well for this use case, feel like it would be a shame to pull it out, but I can "catseye crap" it for our needs if you don't like it.

Reply:
Just because things exist in other parts of the codebase doesn't mean we don't want to improve on them over time. 
```lua
    -- Dynamic modifier that checks party member list on tick to apply
    mob:addListener('COMBAT_TICK', 'MUMOR_CTICK', function(mobArg)
        local sambaDurationBoost = 0
        local party = mobArg:getMaster():getPartyWithTrusts()
        for _, member in pairs(party) do
            if member:getObjType() == xi.objType.TRUST then
                if
                    member:getTrustID() == xi.magic.spell.UKA_TOTLIHN
                then
                    sambaDurationBoost = 10
                end
            end
        end

        -- Always set the boost, even if Uka wasn't found.
        -- This accounts for her being in the party and giving the boost
        -- and also if she dies and the boost goes away.
        mobArg:setMod(xi.mod.SAMBA_DURATION, sambaDurationBoost)
    end)
```

In this example, the listener is firing off every tick, going through the whole party of trusts, looking for the relevant other trust, and _always_ setting the mod. Every tick.

This doesn't have to happen any more often than "the composition of the party has changed". The code _could_ be exactly the same, but the listener could be firing when the composition of the party has changed. That could be a very easy 2-3 line C++ change for you to investigate: Add a new listener trigger when the composition of a party changes, and then hook this logic up to that (and that Mumor block too)

Reply:
> Just because things exist in other parts of the codebase doesn't mean we don't want to improve on them over time.

This is also why we're so hellbent on trying to get things right the 1st time - anytime we don't, inevitably someone cites it as where they copied from. 

Some days feel like we've lost some kind of race to fix something up before someone does.

Reply:
removed this section for now and left a TODO